app/controllers/packages_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PackagesController < ApplicationController
+  def index
+    # Transform query params to what the EBSCO RM-API expects
+    query = URI.encode_www_form(
+      search: params[:q],
+      orderby: params[:orderby] || params[:q] ? 'relevance' : 'packagename',
+      count: params[:count] || 25,
+      offset: params[:offset] || 1
+    )
+
+    # Make the request for packages from the RM API
+    response = rmapi.request(:get, "packages?%{query}" % { query: query })
+
+    if response.ok?
+      render jsonapi: response.data.packagesList.map { |package| Package.new(data: package) },
+             meta: { totalResults: response.data.totalResults }
+    else
+      render jsonapi_errors: response.errors,
+             status: response.code
+    end
+  end
+
+  def show
+    # The package id is a composite of the vendor id since package
+    # resources are nested within vendors
+    vendor_id, package_id = params[:id].split('-')
+    included_resources = params[:include] ? params[:include].split(',') : []
+    pending_requests = {}
+
+    package_path = "vendors/%{vendor_id}/packages/%{package_id}" % {
+      vendor_id: vendor_id || 0,
+      package_id: package_id || 0
+    }
+    pending_requests[:package] = [:get, package_path]
+
+    if included_resources.include?('customerResources')
+      package_titles_path = "#{package_path}/titles?%{query}" % {
+        # Required RM API params
+        query: URI.encode_www_form(
+          search: '',
+          searchfield: 'titlename',
+          orderby: 'titlename',
+          count: 25,
+          offset: 1
+        )
+      }
+      pending_requests[:titles] = [:get,  package_titles_path]
+    end
+
+    # This is a 'compound' request, consisting of multiple calls
+    # to the RMAPI. `request_multi` will process each request
+    # in the pending_requests object and return the responses
+    # in a similar data structure
+    responses = rmapi.request_multi(pending_requests)
```

Comment:
This is a lot of non `HTTP` logic to be in the controller. What can we move into a static method on the `Package` model.

Reply:
@cowboyd Hmm, the only thing I can see that might not be request logic is the RM API specific paths (and the consequential query params). Should we have those path creations in static methods of the respective models?

So `Vendor.path_for(params[:id])`, `Package.path_for(params[:id])`, `Package.path_for_titles(params[:id])`, etc?

That would clean up this method to be more like:
```ruby
def show
  included_resources = params[:include] ? params[:include].split(',') : []
  pending_requests = {}

  # Request the package
  pending_requests[:package] = [:get, Package.path_for(params[:id])]

  # Request customer resources
  if included_resources.include?('customerResources')
    pending_requests[:titles] = [:get, Package.path_for_titles(params[:id])]
  end

  # Make both requests
  response = rmapi.request_multi(pending_requests)

  if responses.ok?
    titles = responses.titles.data.titles rescue []
    render jsonapi: Package.new(data: responses.package.data, titles: titles),
           include: params[:include]
  else
    render jsonapi_errors: responses.errors,
           status: responses.code
end
```

Reply:
When I say HTTP, I mean the HTTP interface of the rails app, Actually making the requests to underlying layers belongs more raturally to the models. E.g.

```ruby
package = Package.find(params[:id], included_resources)
if package.errors?
  render jsonapi_errors: package.errors
else
  render jsonapi: package
end
```

Reply:
So the models should be making the requests then? This is different than what we've been doing so far for the RMAPI stuff. Shall we just merge this then and I can open another PR adjusting all of the controllers for vendors, titles, and packages?

That way this PR is just implementing mostly the same pattern we've been using. And another PR can be concerned with the cleanup of them all.

Reply:
I'm with @wwilsman on this one - i think you're on to something here with moving this logic out of the controller, but where it ends up and what form it takes are questions worthy of their own whiteboard sessions.  for now i'm comfortable with this going in in its current state so that UI side of the equation can keep moving today

Reply:
> So the models should be making the requests then? This is different than what we've been doing so far for the RMAPI stuff. 

This feels more consistent with the way rails apps are typically structured. When you have an active record, it is responsible for making the requests to query and deserialize results over the network, just from a postgres server instead of an RMAPI server.

>Shall we just merge this then and I can open another PR adjusting all of the controllers for vendors, titles, and packages?

Sounds good.